src/System-SessionManager.package/SessionManager.class/instance/snapshot.andQuit..st

```diff
@@ -1,25 +1,13 @@
 snapshot and quit
 snapshot: save andQuit: quit
-	| isImageStarting snapshotResult |
-	ChangesLog default logSnapshot: save andQuit: quit.
-	
-	self currentSession stop: quit.	"Image not usable from here until the session is restarted!"
-	save
-		ifTrue: [
-					snapshotResult := Smalltalk snapshotPrimitive.	"<-- PC frozen here on image file"
-					isImageStarting := snapshotResult == true. ]
-		ifFalse: [ isImageStarting := false ].
-	(quit and: [ isImageStarting not ])
-		ifTrue: [ Smalltalk quitPrimitive ].
-
-	"create a new session object if we're booting"
-	isImageStarting ifTrue: [ self installNewSession ].
-
-	self currentSession start: isImageStarting.
-	snapshotResult 
-		ifNil: [ self error: 'Failed to write image file (disk full?)' ]
-		ifNotNil: [ 
-			SystemAnnouncer uniqueInstance snapshotDone: isImageStarting ].
-	
-	"We return the resuming state, which may be useful for users to know the state of the image"
+	| isImageStarting wait |
+	"We do the snapshot in a separate process in maximum priority to have always a clean startup.
+	This process will be interrupted by the fork, and will be resumed as soon as the snapshot finishes.
+	We synchronize these processes in case both are in the same priority"
+	wait := Semaphore new.
+	[ 
+		isImageStarting := self launchSnapshot: save andQuit: quit.
+		wait signal
+	] forkAt: Processor highestPriority.
+	wait wait.
```

Comment:
What if the new process dies before wait signal is called? Should this be in an ensure block?